### ISSUE TITLE
Error: Penthouse timed out after 30s.

### ISSUE THREAD

**Issue by danchukas** [addyosmani/critical#284]
`cat /tmp/cs_4DTK26 | critical --folder /project_public_root --inline > /tmp/cd_TKmFq3` 
Error: Penthouse timed out after 30s. 

**/tmp/cs_4DTK26:**
```
<!DOCTYPE html>
<html>
<head>
    <link rel="preload" href="https://project.com.local/template/css/cache/modstyles_ca56d2404478ebb290ec8f47c830bd02_1516027304.css_v_1516027304" as="style">
...
```

**/tmp/cd_TKmFq3** is empty

What to do ?

**Comment by pocketjoso** [comment]
I think you should be able to change the default penthouse timeout from critical. The critical library accepts a `penthouse` options object, you can set it to: `{timeout: 60000}` to see if that resolves your issue.

If your still get timeouts, then something on your site seems to be causing Penthouse to hang. If so, can you share a public url that you are generating critical css for, and I can test it?

**Comment by situ185** [comment]
I have testing this module on a couple of sites but i keep getting the timeout on www.hertz.com.
I have tried increasing the penthouse timeout to 120000 but still keeps timing out.
It works on www.msn.com and www.yahoo.com.
Could you take a look @pocketjoso 
I get a Unhandled rejection Error.
call stack is 

_penthouse\lib\core.js:217:20)
at ontimeout (timers.js:498:11)
at tryOnTimeout (timers.js:323:5)
at Timer.listOnTimeout (timers.js:290:5)_

thankyou

**Comment by epcliff** [comment]
Hi, I have the same issue. It was working before and now I suddenly get this penthouse error message.

**Comment by pocketjoso** [comment]
As I mentioned earlier, it's most likely one of two things that is happening.

## Job is really running out of time
If you are running critical/penthouse for lots of pages/urls/dimensions at the same time, then it's possible that your computer is just slowing down because too much execution is done in parallel. You can try reducing the number of critical css files you are trying to execute at the same time, and/or increasing the timeout.

## Job is hanging
In this case it's most likely something specific on the page that causes the problem. To help with this, I need the specific URL you are using so that I can debug.

@danchukas I'm sorry, I think you replied to me before, but somehow I _only_ got an email, and I didn't see your reply here.. It was mentioned that hertz.com causes timeouts like this in Penthouse. What causes the problem on hertz.com is this code in the html:
```html
<meta HTTP-EQUIV="REFRESH" content="0; url=/rentacar/errors/jsCookieErrorPage?nojs=true">
```
This causes the page to unload during the critical css generation, which is not supported. In earlier versions of Penthouse this would cause penthouse to hang and thus timeouts, but this was changed in [Penthouse@1.6.1](https://github.com/pocketjoso/penthouse/releases/tag/v.1.6.1), which the latest Critical version uses. Instead an error is thrown.

**Comment by mems** [comment]
I think it's related to a bug in penthouse: https://github.com/pocketjoso/penthouse/issues/274
And it has been fixed in [penthouse@1.9.0](https://github.com/pocketjoso/penthouse/releases/tag/v1.9.0)

**Comment by bezoerb** [comment]
Closing this as the issue should be fixed in penthouse. If there's no `meta refresh` or anything else that could cause a page redirect on load you can also try changing the penthouse timeout.
